tests/test_redirection.rs

```diff
@@ -1,37 +1,43 @@
+use std::net::TcpListener;
 use std::net::TcpStream;
 use std::thread;
 use std::time::{Duration, Instant};
 
 use attohttpc::ErrorKind;
 use lazy_static::lazy_static;
-use rouille::Response;
+use rouille::{router, Response};
 
 lazy_static! {
-    static ref STARTED: bool = {
+    static ref STARTED: u16 = {
+        let port = TcpListener::bind("localhost:0").unwrap().local_addr().unwrap().port();
```

Comment:
Isn't this racy without explicitly setting `SO_REUSE` port?

Reply:
Is there anyway to provide rouille with the TcpListener to use? I think this might be racy yes.

Reply:
> Is there anyway to provide rouille with the TcpListener to use? I think this might be racy yes.

The default recipe I know of is to bind using `SO_REUSEPORT` and keep the initial listener alive while starting the second one but dropping it before establishing connections. But honestly speaking, I think adamreichold@580abf0 is much simpler.

Reply:
Last commit did resolve the race. @tdryer I think it should also allow us to remove the waiting loop since we only spawn the thread after the socket is bound?

Reply:
Yep, the waiting loop is unnecessary now. I'll remove it.